### DISCUSSION TITLE
Zigbee 2 Tasmota Router Issue

### DISCUSSION THREAD

**Discussion by zizebra** [arendst/Tasmota#13460] [Q&A] [answered]
Hi all, can someone guide on my current issue. I have a Z2T Sonoff Bridge  which has been working flawlessly for the last 8months.  
My main challenge is on the routers work with Z2T.  I have 2 Xiaomi bulbs which doubles up routers. I have noted that whenever some of the sensors connect through a router, their identities are changed and their statuses are published using new device IDs. Is there something i need to do to make Z2T to retain the original IDs 

My deployment has the following 
1 X Sonoff ZBBridge flashed with Z2T
2 XSamsung Water Leak Sensors
2 X Xiaomi Bulbs
2 x Sonoff Door Sensors
2 x Motion Sensors
<img width="622" alt="ZB" src="https://user-images.githubusercontent.com/4173129/138609679-8909d9ed-4423-4b89-8ca9-fc8473c92acc.PNG">

**Comment by s-hadinger** [comment, marked as answer]
No. It's all done at device level. Some have strange behaviors. If you use friendly names they are updated automatically so it shouldn't have much impact

**Reply by zizebra** [reply]
thank you. Are you saying, it is expected that the device ID to change? Iam using friendlynames, in the screenshot of the map posted above, geyserleaksensor already had an ID 0x8b32, and since  4days ago it is being recieved through the router using a new device id 0x50A1 also shown in the diagram.

**Reply by sfromis** [reply]
If you just do not use that short address for anything at all, this strangeness should not cause trouble.

**Reply by zizebra** [reply]
Well i am not using any short address for anything,i have never used for anything at all. i just noted that a couple of rules stopped working and upon investigation i found that all devices whose rules were not working seemed to have changed. The only reason for this behavior seems that the routing changed.  So far all i can see is that i have to pair them again, but it would be good to know why this is happening

**Reply by zizebra** [reply]
Since the last time I posted. The devices which I reregistered have recieved new IDs whenever they switch the router they are connecting with.

**Reply by ascillato2** [reply]
Are you using the command ZbName to set a friendly name to your device?
The IEEE address never change, but the short address does change on re-pairing. That is why, it is recommended to set a friendly name. Friendly names don't change due to they are linked to their IEEE address.

**Reply by zizebra** [reply]
Iam probably missing something because you keep asking if iam setting friendly name.  Iam using zbname to set the friendly names. Is there another way to set friendly name without using zbname. As you can see now its a mess.. yesterday i decided to add the new IDs using existing friendly names. Today there is new IDs again. I probably have to reset and start all over. But i would be happy to reset  after someone has shed somelight on why this is happening.
![image](https://user-images.githubusercontent.com/4173129/139326671-dbf4f1bd-41f9-4ff5-a9c1-58e11d291691.png)
![image](https://user-images.githubusercontent.com/4173129/139326821-d79d815c-92f9-4526-9da3-98f73b8fcada.png)

**Reply by zizebra** [reply]
For one of the sensors which keeps getting changed here is the info for all the status IDs which got created for one item

{"ZbStatus2":[
{"Device":"0xB832",
"Name":"GeyserLeakSensor",
"IEEEAddr":"0x286D9700010758B5",
"ModelId":"water",
"Manufacturer":"Samjin",
"Endpoints":[1],
"Config":["A01.4","T01"]}]}

{"ZbStatus2":[
{"Device":"0x50A1",
"Name":"GeyserLeakSensor",
"IEEEAddr":"0x0000000000000000",
"Endpoints":[1],
"Config":["A01","T01"]}]}

{"ZbStatus2":[
{"Device":"0x30B9",
"Name":"GeyserLeakSensor",
"IEEEAddr":"0x0000000000000000",
"Endpoints":[1],
"Config":["A01","T01"]}]}

{"ZbStatus2":[
{"Device":"0xAFEA",
"IEEEAddr":"0x0000000000000000",
"Endpoints":[1],
"Config":["T01","A01"]}]}

**Reply by ascillato** [reply]
The "IEEEAddr":"0x0000000000000000" is bad. It shouldn't be there.

Please, update your device to latest Tasmota from the development branch ( http://ota.tasmota.com/tasmota/ ) and then update again the ZigBee Module of your bridge. Then do a `reset 1` from the console to start from scratch. You will need to pair all devices again.

**Reply by zizebra** [reply]
ok i will reset this evening. Iam running the latest 

Program Version | 10.0.0(zbbridge)
2021-10-19T08:23:51
2_7_4_9/2.2.2-dev(38a443e)

**Reply by s-hadinger** [reply]
Those devices have a wrong behavior and they seem not to announce themselves when reconnecting. One thing we can try is to force a `ZbProbe ` whenever a new address arrives, but may break other stuff

**Reply by zizebra** [reply]
my joy was short-lived. I thought the reset had done it.  I have however noted that this is happening when we lose power in whole house. We have been experiencing power cuts. apparently power cuts stopped on Saturday and resumed today. Then today the some devices started showing up with new IDs again.

**Reply by zizebra** [reply]
one more thing reset 1 did not erase the everything, the sonoff wifi was reset but the zigbee devices were still remembered.

**Reply by zizebra** [reply]
zbreset 1 did erase everything on ZB
